### ISSUE TITLE
Cronline#next_time hangs on crossing dst boundary wiht ActiveSupport::TimeWithZone

### ISSUE THREAD

**Issue by fsateler** [codez/delayed_cron_job#30]
The America/Santiago timezone switches to DST on Saturday 4th of September 2021. THis causes a hang when attempting to get the next time while crossing the DST boundary:

```ruby
# Demonstrates the DST Crossing problem
require 'active_support'
require 'delayed_cron_job'

Time.zone = 'America/Santiago'

Timeout.timeout(10) { DelayedCronJob::Cronline.new('* * * * 1 ').next_time(Time.utc(2021, 9, 1)) }
puts "Time with utc works!"

Timeout.timeout(10) { DelayedCronJob::Cronline.new('* * * * 1 ').next_time(Time.new(2021, 9, 1, 0, 0, 0, TZInfo::Timezone.get('America/Santiago'))) }
puts "Time with timezone works!"

Timeout.timeout(10) { DelayedCronJob::Cronline.new('* * * * 1 ').next_time(Time.new(2021, 9, 1, 0, 0, 0, '-04')) }
puts "Time with timezone offset works!"

Timeout.timeout(10) { DelayedCronJob::Cronline.new('* * * * 1 ').next_time(Time.zone.local(2021, 1, 1)) }
puts "ActiveSupport::Timezone without dst crossing works!"

Timeout.timeout(10) { DelayedCronJob::Cronline.new('* * * * 1 ').next_time(Time.zone.local(2021, 9, 1)) }
# This is never reached :(
puts "ActiveSupport::Timezone with dst crossing works!"
```

The last line fails. 

Instrumenting `next_time` to print the dates as it moves them:

```ruby
        unless date_match?(time)
          otime = time
          dst = time.isdst
          time += (24 - time.hour) * 3600 - time.min * 60 - time.sec
          time -= 3600 if time.isdst != dst # not necessary for winter, but...
          puts time
          if time == otime
            raise 'time didnt move!'
          end
          next
        end
```

Gives the following output:

```
2021-09-02 00:00:00 UTC
2021-09-03 00:00:00 UTC
2021-09-04 00:00:00 UTC
2021-09-05 00:00:00 UTC
2021-09-06 00:00:00 UTC
Time with utc works!
2021-09-02 00:00:00 -0400
2021-09-03 00:00:00 -0400
2021-09-04 00:00:00 -0400
2021-09-05 01:00:00 -0300
2021-09-06 00:00:00 -0300
Time with timezone works!
2021-09-02 00:00:00 -0400
2021-09-03 00:00:00 -0400
2021-09-04 00:00:00 -0400
2021-09-05 01:00:00 -0300
2021-09-06 00:00:00 -0300
Time with timezone works!
2021-09-02 00:00:00 -0400
2021-09-03 00:00:00 -0400
2021-09-04 00:00:00 -0400
2021-09-05 00:00:00 -0400
2021-09-06 00:00:00 -0400
Time with timezone offset works!
2021-01-02 00:00:00 -0300
2021-01-03 00:00:00 -0300
2021-01-04 00:00:00 -0300
ActiveSupport::Timezone without dst crossing works!
2021-09-02 00:00:00 -0400
2021-09-03 00:00:00 -0400
2021-09-04 00:00:00 -0400
2021-09-04 23:00:00 -0400
2021-09-04 23:00:00 -0400
Traceback (most recent call last):
        9: from test-delayed.rb:21:in `<main>'
        8: from /home/felipe/.rbenv/versions/2.7.4/lib/ruby/2.7.0/timeout.rb:110:in `timeout'
        7: from /home/felipe/.rbenv/versions/2.7.4/lib/ruby/2.7.0/timeout.rb:33:in `catch'
        6: from /home/felipe/.rbenv/versions/2.7.4/lib/ruby/2.7.0/timeout.rb:33:in `catch'
        5: from /home/felipe/.rbenv/versions/2.7.4/lib/ruby/2.7.0/timeout.rb:33:in `block in catch'
        4: from /home/felipe/.rbenv/versions/2.7.4/lib/ruby/2.7.0/timeout.rb:95:in `block in timeout'
        3: from test-delayed.rb:21:in `block in <main>'
        2: from /home/felipe/src/satelabs/buk/gems/ruby/2.7.0/gems/delayed_cron_job-0.7.4/lib/delayed_cron_job/cronline.rb:131:in `next_time'
        1: from /home/felipe/src/satelabs/buk/gems/ruby/2.7.0/gems/delayed_cron_job-0.7.4/lib/delayed_cron_job/cronline.rb:131:in `loop'
/home/felipe/src/satelabs/buk/gems/ruby/2.7.0/gems/delayed_cron_job-0.7.4/lib/delayed_cron_job/cronline.rb:139:in `block in next_time': asdf (RuntimeError)
```

It looks to me as if the `time.isdst` check should change sign depending on whether we are entering or exiting DST, but I'm not sure I understand the logic correctly. Interestingly, the native ruby Time never triggers the `isdst` check (`Time.new(2021, 9, 5, 1, 0, 0, TZInfo::Timezone.get('America/Santiago')).isdst` is still false).

**Comment by codez** [comment]
Thank you for reporting the issue. Seems like the cron algorithm is not entirely solved yet ;)

Probably relates to https://github.com/floraison/fugit/issues/60
I will look into using Fugit directly or inline its Cron class, as this is the source for our Cronline class anyways.

**Comment by codez** [comment]
I released version 0.8.0 depending on the fixed fugit version.
